swiftMongoDB/MongoClient.swift

```diff
@@ -56,29 +56,34 @@ public class MongoClient {
         }
     }
     
-
     // authenticated connection - required specific database
-    public init(host: String, port: Int, database: String, usernameAndPassword: (username: String, password: String)) throws {
+    public convenience init(host: String, port: Int, database: String, usernameAndPassword: (username: String, password: String)) throws {
+        try self.init(
+            host: host,
+            port: port,
+            database: database,
+            authenticationDatabase: database,
+            usernameAndPassword: usernameAndPassword
+        )
+    }
+    
+    // authenticated connection - required specific database and specific database for authentication
+    public init(host: String, port: Int, database: String, authenticationDatabase: String, usernameAndPassword: (username: String, password: String)) throws {
     
         let userAndPass = "\(usernameAndPassword.username):\(usernameAndPassword.password)@"
         
-        self.clientURI = "mongodb://\(userAndPass)\(host):\(port)/?authSource=\(database)"
+        self.clientURI = "mongodb://\(userAndPass)\(host):\(port)/\(database)?authSource=\(authenticationDatabase)"
         
         self.host = host
         self.port = port
         
-        self.clientRAW = mongoc_client_new(self.clientURI)
-        
         mongoc_init()
         
-        let reply = bson_new()
-        var status = bson_error_t()
-        mongoc_client_get_server_status(self.clientRAW, nil, reply, &status)
-        
+        self.clientRAW = mongoc_client_new(self.clientURI)
         
-        if status.code.mongoError != MongoError.NoError {
-        print(errorMessageToString(&status.message))
-        throw status.code.mongoError
+        if (self.clientRAW == nil) {
+            print("Invalid hostname or port: %s\n", self.host+" \(+self.port)");
```

Comment:
What about using `\()` instead of `%s` and string concatenation? Also, another comment about this on the next commit :)